src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -187,42 +190,43 @@ internal static ITypeSymbol GetNullableUnderlyingType(ITypeSymbol type)
             return ((INamedTypeSymbol)type).TypeArguments[0];
         }
 
-        internal static bool IsBooleanType(ITypeSymbol type)
+        internal static bool IsBooleanType([NotNullWhen(returnValue: true)] ITypeSymbol? type)
         {
             return type?.SpecialType == SpecialType.System_Boolean;
         }
 
-        internal static bool IsObjectType(ITypeSymbol type)
+        internal static bool IsObjectType([NotNullWhen(returnValue: true)] ITypeSymbol? type)
         {
             return type?.SpecialType == SpecialType.System_Object;
         }
 
-        internal static bool IsSignedIntegralType(ITypeSymbol type)
+        internal static bool IsSignedIntegralType([NotNullWhen(returnValue: true)] ITypeSymbol? type)
         {
             return type?.SpecialType.IsSignedIntegralType() == true;
         }
 
-        internal static bool IsUnsignedIntegralType(ITypeSymbol type)
+        internal static bool IsUnsignedIntegralType([NotNullWhen(returnValue: true)] ITypeSymbol? type)
         {
             return type?.SpecialType.IsUnsignedIntegralType() == true;
         }
 
-        internal static bool IsNumericType(ITypeSymbol type)
+        internal static bool IsNumericType([NotNullWhen(returnValue: true)] ITypeSymbol? type)
         {
             return type?.SpecialType.IsNumericType() == true;
         }
 
-        internal static ITypeSymbol GetEnumUnderlyingType(ITypeSymbol type)
+        internal static ITypeSymbol? GetEnumUnderlyingType(ITypeSymbol? type)
```

Comment:
>? [](start = 70, length = 1)

I'd be tempted to restrict input here (API not really useful with null input) as with `GetMembers` (discussed above)

Reply:
I'll leave this one as-is since the API is internal, and the change wouldn't provide a direct benefit to consumers. If someone wants to clean this up for some reason they will be able to do so as a separate pull request.